docs/whats-new/preview-4.md

```diff
@@ -0,0 +1,699 @@
+var storage = builder.AddAzureStorage("storage").RunAsEmulator(container =>
+{
+    container.UsePersistence();
+});
+```
+
+**CosmosDB emulator**
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var db = builder.AddAzureCosmosDB("cosmos")
+                .RunAsEmulator()
+                .AddDatabase("db");
+
+builder.AddProject<Projects.Api>("api")
+       .WithReference(db);
+
+builder.Build().Run();
+```
+
+## The deployment manifest
+
+The [.NET Aspire manifest](../deployment/manifest-format.md) has undergone a significant overhaul to support the new primitives and introduced in preview 4. The new manifest focuses on a few key primitives to enable a wide range of scenarios:
+
+| Resource type   | Description                                                   |
+|-----------------|---------------------------------------------------------------|
+| `project.v0`    | .NET project files                                            |
+| `container.v0`  | Container images                                              |
+| `dockerfile.v0` | Docker files                                                  |
+| `parameter.v0`  | External Parameters                                           |
+| `value.v0`      | References to other resources (or a combination of resources) |
+
+### Azure specific resources
+
+| Resource type    | Description           |
+|------------------|-----------------------|
+| `azure.bicep.v0` | Azure Bicep templates |
+
+Tool authors can support this very small set of resource types to model lots of different apps!
+
+We've deprecated the abstract resource types that were supported in previous versions. Deployment tools may still support those resource types, but they are no longer part of the core manifest schema.
+
+### Prompting for parameters
+
+When azd detects a parameter.v0 resource in the manifest, it will prompt for a value. For example, consider the following code in the apphost project:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var administratorLogin = builder.AddParameter("administratorLogin");
+var administratorLoginPassword = builder.AddParameter(
+    "administratorLoginPassword", secret: true);
+
+var pg = builder.AddPostgres("postgres")
+                .AsAzurePostgresFlexibleServer(
+                    administratorLogin, administratorLoginPassword)
+                .AddDatabase("db");
+```
+
+In the preceding code, two parameters are created to represent the admin login and password. They're then passed to the `AsAzurePostgresFlexibleServer` API. This causes `azd` to ask for both parameter values.
+
+:::image type="content" source="media/preview-4/promptUser.gif" lightbox="media/preview-4/promptUser.gif" alt-text="azd parameters prompting":::
+
+To skip interactive prompting, users can set an environment variable for each parameter. For the `administratorLogin` parameter, azd looks for an environment variable with the name `AZURE_ADMINISTRATOR_LOGIN`.
```

Comment:
Is this correct?

Reply:
I see a test that includes it, but nowhere else in the source:

https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/apphost/testdata/TestAspireBicepGeneration-main.parameters.json.snap#L9